lib/index.js

```diff
@@ -146,16 +148,26 @@ Appboy.prototype.initializeV2 = function(customEndpoint) {
   };
 
   if (customEndpoint) config.baseUrl = customEndpoint;
-  
+
   this.initializeTester(options.apiKey, config);
   window.appboy.initialize(options.apiKey, config);
-  
-  if (options.automaticallyDisplayMessages) window.appboy.display.automaticallyShowNewInAppMessages();
-  if (userId) window.appboy.changeUser(userId);
-  
-  window.appboy.openSession();
 
-  this.load('v2', this.ready);
+  if (options.automaticallyDisplayMessages) window.appboy.display.automaticallyShowNewInAppMessages();
+  if (userId) {
```

Comment:
@ladanazita @ucarion @viparthasarathy try inverting the userId check and load:

```
  self.load('v2', function() {
    var userId = self.analytics.user().id();
    if (userId) {
      window.appboy.changeUser(userId, function() {
        window.appboy.openSession(function() {
          self.ready();
        });
      });
    } else {
      window.appboy.openSession(function() {
        self.ready();
      });
    }
  });
};
```
(I think this was the main reason the test earlier didn't work as expected)

Reply:
Going to assume this isn't necessary anymore as we'd want to keep the `.identify` call inside of the `.ready` callback anyways with the suggested global callback solution.